### PR TITLE
Vec: Use `NTuple` instead of `SVector`

### PR DESCRIPTION
**Pull request by eliascarv**

### CODE REVIEW

**Comment by juliohm** [comment]
Thank you @eliascarv for working on this tricky changes. Can you please confirm that there are no performance issues compared to the master branch? I am looking forward to dropping SVector from more places in the codebase.

File: src/vectors.jl [hunk 1]
```diff
@@ -44,22 +44,31 @@ Vec3f(1, 2, 3) # explicitly ask for single precision
...
 # convenience constructors
-Vec{Dim,T}(coords...) where {Dim,T} = Vec{Dim,T}(SVector{Dim,T}(coords))
-Vec{Dim,T}(coords::Tuple) where {Dim,T} = Vec{Dim,T}(SVector{Dim,T}(coords))
-Vec{Dim,T}(coords::AbstractVector) where {Dim,T} = Vec{Dim,T}(SVector{Dim,T}(coords))
+Vec{Dim,T}(coords...) where {Dim,T} = Vec{Dim,T}(coords)
+function Vec{Dim,T}(coords::Tuple) where {Dim,T}
+  checkdim(Vec{Dim,T}, coords)
+  Vec{Dim,T}(NTuple{Dim,T}(coords))
+end
+
```
Comment (juliohm) [inline src/vectors.jl]:
```suggestion
```

File: src/vectors.jl [hunk 2]
```diff
@@ -70,10 +78,23 @@ const Vec2f = Vec{2,Float32}
...
-Base.getindex(v::Vec, i::Int) = getindex(v.coords, i)
+Base.Tuple(v::Vec) = getfield(v, :coords)
+Base.getindex(v::Vec, i::Int) = getindex(getfield(v, :coords), i)
 function StaticArrays.similar_type(::Type{<:Vec}, ::Type{T}, ::Size{S}) where {T,S}
   L = prod(S)
   N = length(S)
-  N == 1 ? Vec{L,T} : SArray{Tuple{S...}, T, N, L}
+  N == 1 ? Vec{L,T} : SArray{Tuple{S...},T,N,L}
+end
+
```
Comment (juliohm) [inline src/vectors.jl]:
```suggestion
```

**Comment by juliohm** [comment]
@eliascarv is this ready to be merged?

**Comment by eliascarv** [comment]
Yes, it's ready!

**Comment by juliohm** [comment]
Amazing work @eliascarv as usual! 💯

**Pull request merged**